include/gridtools/common/hymap.hpp

```diff
@@ -162,8 +162,24 @@ namespace gridtools {
 
         template <class Key, class Map>
         using element_at = tuple_util::element<meta::st_position<get_keys<Map>, Key>::value, Map>;
+
+        template <class T, class Keys = get_keys<T>>
+        struct is_hymap : bool_constant<meta::is_set<Keys>::value && meta::length<Keys>::value == tuple_util::size<T>::value> {};
     } // namespace hymap_impl_
 
+    template <class, class = void>
+    struct is_hymap : std::false_type {};
+
+    template <class T>
+    struct is_hymap<T, std::enable_if_t<hymap_impl_::is_hymap<T>::value>> : is_tuple_like<T> {};
```

Comment:
Why did you decide to place the is_tuple_like here, instead of impl_. It might be confusing if you are looking up is_hymap and find line 167 by search.

Reply:
I have re-implemented the predicate. I think it is less confusing now.